src/main/scala/com/github/vitalsoftware/macros/JsonFormatAnnotation.scala

```diff
@@ -48,31 +48,53 @@ class jsonMacro(useDefaults: Boolean) {
       }
     }
 
+    /**
+     * Generates implicit json.Formats that fallback to the defaults is the initial parsing fails.
+     */
     def jsonFormatter(className: TypeName, fields: List[ValDef]) = {
       fields.length match {
         case 0 => c.abort(c.enclosingPosition, "Cannot create json formatter for case class with no fields")
         case _ =>
-          if (useDefaults) {
-            q"implicit val jsonAnnotationFormat = play.api.libs.json.Json.using[play.api.libs.json.Json.WithDefaultValues].format[$className]"
+          val lowPriorityFormats = if (useDefaults) {
+            q"private val lowPriorityFormats = play.api.libs.json.Json.using[play.api.libs.json.Json.WithDefaultValues].format[$className]"
           } else {
-            q"implicit val jsonAnnotationFormat = play.api.libs.json.Json.format[$className]"
+            q"private val lowPriorityFormats = play.api.libs.json.Json.format[$className]"
           }
+
+          Seq(
+            lowPriorityFormats,
+            q"""
+                private val robustReads = new play.api.libs.json.Reads[$className] {
+                  override def reads(json: play.api.libs.json.JsValue) = lowPriorityFormats.reads(json) match {
+                    case s: play.api.libs.json.JsSuccess[_] => s
+                    case e: play.api.libs.json.JsError =>
+                      val transforms = e.errors.map(_._1)
+                        .filter(_.path.size == 1)
+                        .map(_.json.prune)
+                        .reduce(_ andThen _)
+
+                      json.transform(transforms).flatMap(lowPriorityFormats.reads _).orElse(e)
```

Comment:
The size check is for the depth of the Json path. I wanted to avoid replacing any deep values as they will get anyway processed recursively when `Json.Reads` works on a object.